devito/propagator.py

```diff
@@ -47,11 +50,12 @@ class Propagator(object):
                            Set cache_blocking to AutoTuner instance, to use auto tuned
                            tune block sizes
     """
-    def __init__(self, name, nt, shape, stencils, factorized=None, spc_border=0,
-                 time_order=0, time_dim=None, space_dims=None, dtype=np.float32,
-                 forward=True, compiler=None, profile=False,
+    def __init__(self, name, nt, shape, stencils=None, loop_body=None, factorized=None,
```

Comment:
I moved it to a symbolic only implementation. Let's leave the lower-level API for another time.